src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -34,12 +35,19 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
             const templates = this.cftRegistry.registeredTemplates
 
             for (const templateDatum of templates) {
-                if (templateDatum.path.startsWith(folderPath) && templateDatum.template.Resources) {
+                const folderPathPieces = folderPath.split(sep)
```

Comment:
Move all the logic to support testing if a file is within a folder into a utility method
Motivations:
- reusable (and testable)
- easier to understand the logic that this function is performing

Reply:
Done, moved to filesystemUtils